src/vmod_xkey.c

```diff
@@ -500,65 +500,77 @@ purge(VRT_CTX, VCL_STRING key, VCL_INT do_soft)
 	unsigned char digest[DIGEST_LEN];
 	struct xkey_hashhead *hashhead;
 	struct xkey_oc *oc;
-	int i;
+	int i = 0;
+	const char *ep, *sp;
 
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
 	CHECK_OBJ_NOTNULL(ctx->req, REQ_MAGIC);
 	CHECK_OBJ_NOTNULL(ctx->req->wrk, WORKER_MAGIC);
 
 	if (!key || !*key)
 		return (0);
-
-	SHA256_Init(&sha_ctx);
-	SHA256_Update(&sha_ctx, key, strlen(key));
-	SHA256_Final(digest, &sha_ctx);
-
-	AZ(pthread_mutex_lock(&mtx));
-	hashhead = xkey_hashtree_lookup(digest, sizeof(digest));
-	if (hashhead == NULL) {
-		AZ(pthread_mutex_unlock(&mtx));
-		return (0);
-	}
-	i = 0;
-	VTAILQ_FOREACH(oc, &hashhead->ocs, list_hashhead) {
-		CHECK_OBJ_NOTNULL(oc->objcore, OBJCORE_MAGIC);
-		if (oc->objcore->flags & OC_F_BUSY)
-			continue;
+	sp = key;
+	while (*sp != '\0') {
+		while (*sp == ' ')
+			sp++;
+		ep = sp;
+		while (*ep != '\0' && *ep != ' ')
+			ep++;
+		if (sp == ep)
```

Comment:
Thanks @Dridi I do agree it is confusing however can't take credit for this code: it was copied from another part of the vmod: https://github.com/varnish/varnish-modules/blob/master/src/vmod_xkey.c#L416

It seemed like a good idea to keep the key-splitting code in both places consistent, if I update my PR should I also make the same change in xkey_cb_insert()?

I'm happy to incorporate any further suggestions into my pull request.

Regards